fast64_internal/f3d/f3d_material.py

```diff
@@ -2440,6 +2442,61 @@ def reloadDefaultF3DPresets():
             update_preset_manual_v4(material, presetNameToFilename[material.f3d_mat.presetName])
 
 
+class UpdateColorSpacePopup(Operator):
+    bl_label = "Update Color Space"
+    bl_idname = "dialog.update_colorspace"
+    bl_description = "Update color space"
+    bl_options = {"UNDO"}
+
+    already_invoked = False  # HACK: used to prevent multiple dialogs from popping up
+
+    def invoke(self, context: Context, event: Event):
+        if UpdateColorSpacePopup.already_invoked:
+            return {"FINISHED"}
+        scene = context.scene
+        view_settings = scene.view_settings
+        # Check if color space is correct
+        if not scene.dont_ask_colorspace and (
+            scene.display_settings.display_device != "sRGB"
+            or view_settings.view_transform != "Standard"
+            or view_settings.look != "None"
+            or view_settings.exposure != 0.0
+            or view_settings.gamma != 1.0
+        ):
+            UpdateColorSpacePopup.already_invoked = True
+            return context.window_manager.invoke_props_dialog(self, width=400)
+
+        return {"FINISHED"}
+
+    def draw(self, context: Context):
+        col = self.layout.column()
+        multilineLabel(
+            col,
+            f'The colorspace for the current scene "{context.scene.name}"\nwill cause inaccurate preview compared to N64.\nWould you like to update it?',
+            icon="INFO",
+        )
+        col.prop(context.scene, "dont_ask_colorspace")
+
+    def cancel(self, context: Context):
+        UpdateColorSpacePopup.already_invoked = False
+
+    def execute(self, context):
+        try:
+            scene = context.scene
+            scene.display_settings.display_device = "sRGB"
+            scene.view_settings.view_transform = "Standard"
+            scene.view_settings.look = "None"
+            scene.view_settings.exposure = 0.0
+            scene.view_settings.gamma = 1.0
+            self.report({"INFO"}, "Updated color space.")
+            return {"FINISHED"}
+        except Exception as exc:
+            raisePluginError(self, exc)
+            return {"CANCELLED"}
```

Comment:
```suggestion
```
(just don't handle the exception if all that's done is (poorly) reraising it)

Reply:
the point is to report and print traceback, returning CANCELLED also means that no undo entry is added, its done every where else in fast64 so why should it not be done here exactly?

Reply:
My bad, I thought raisePluginError raised an exception but it doesn't, only reports it

idk about not wanting to register an undo step, in theory props could have been changed by the operator by then